app/assets/javascripts/index/directions-route-output.js

```diff
@@ -105,22 +126,37 @@ OSM.DirectionsRouteOutput = function (map) {
           map.removeLayer(highlight);
         });
     }
+  }
+
+  const routeOutput = {};
+
+  routeOutput.write = function (route) {
+    polyline
+      .setLatLngs(route.line)
+      .addTo(map);
+
+    writeSummary(route);
+    writeSteps(route);
+
+    $("#directions_distance_units_km").off().on("change", () => {
+      distanceUnits = "km";
+      writeSummary(route);
+      writeSteps(route);
+    });
+    $("#directions_distance_units_mi").off().on("change", () => {
+      distanceUnits = "mi";
+      writeSummary(route);
+      writeSteps(route);
+    });
 
     const blob = new Blob([JSON.stringify(polyline.toGeoJSON())], { type: "application/json" });
     URL.revokeObjectURL(downloadURL);
     downloadURL = URL.createObjectURL(blob);
+    $("#directions_route_download").prop("href", downloadURL);
```

Comment:
If you have the DOM anyway you can drop `downloadURL`

```suggestion
    URL.revokeObjectURL(downloadLink.href);
    downloadLink.href = URL.createObjectURL(blob);
```

Reply:
And it can do the job of `downloadURL` too since you preserve the DOM starting with 9e908a00b0f7d7c4ffbbf93b8dae80a3d2c3b05f

Reply:
I'm saying that it's not a string variable that preserves the value you put in it.